### PR TITLE
Change API from SortedMap to NavigableMap.

### PR DESCRIPTION
**Pull request by jkozlowski**
**Goals (and why)**:

Latest baseline complains about usages of SortedMap.

**Implementation Description (bullets)**:

**Testing (What was existing testing like?  What have you done to improve it?)**:

**Concerns (what feedback would you like?)**:

**Where should we start reviewing?**:

**Priority (whenever / two weeks / yesterday)**:

### CODE REVIEW

File: atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/Cells.java [hunk 1]
```diff
@@ -111,7 +112,7 @@ public static Cell createLargestCellForRow(byte[] rowName) {
         return ret;
     }
 
-    public static <T> NavigableMap<byte[], SortedMap<byte[], T>> breakCellsUpByRow(Map<Cell, T> map) {
+    public static <T> NavigableMap<byte[], NavigableMap<byte[], T>> breakCellsUpByRow(Map<Cell, T> map) {
```
Comment (jeremyk-91) [inline atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/Cells.java:115]: This one is potentially dev-breaky (generics are invariant), though a quick look internally suggests that it has a few users outside AtlasDB, the last of which was updated in mid-2019 so I think it's fine. I'd suggest calling this one out in changelog to be certain.

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/ResultsExtractor.java [hunk 2]
```diff
@@ -78,15 +78,15 @@ public ResultsExtractor(MetricsManager metricsManager) {
...
+        NavigableMap<byte[], NavigableMap<byte[], T>> resultsByRow = Cells.breakCellsUpByRow(asMap());
         return getRowResults(endExclusive, lastRow, resultsByRow);
     }
 
     public static <T> TokenBackedBasicResultsPage<RowResult<T>, byte[]> getRowResults(
             byte[] endExclusive,
             byte[] lastRow,
-            SortedMap<byte[], SortedMap<byte[], T>> resultsByRow) {
-        SortedMap<byte[], RowResult<T>> ret = RowResults.viewOfSortedMap(resultsByRow);
+            NavigableMap<byte[], NavigableMap<byte[], T>> resultsByRow) {
```
Comment (jeremyk-91) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/ResultsExtractor.java:88]: This one doesn't seem to have external users, so should be fine

File: atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/Cells.java (hunk 1 above)
Comment (jkozlowski) [reply atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/Cells.java:114]: Added to the changelog.

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/ResultsExtractor.java (hunk 2 above)
Comment (jkozlowski) [reply atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/ResultsExtractor.java:88]: Still added to changelog.

File: atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java [hunk 3]
```diff
@@ -65,7 +65,7 @@
      * @return a mapping of rows to the columns matching the provided column selection
      */
     @Idempotent
-    SortedMap<byte[], RowResult<byte[]>> getRows(
+    NavigableMap<byte[], RowResult<byte[]>> getRows(
```
Comment (j-baker) [inline atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java:68]: isn't this a binary backcompat break?

Comment (iamdanfox) [reply atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java:68]: Yes, just hit this. It seems that anyone relying on internal libraries (like atlas-event-queue) will need to upgrade as they get errors like this:

```
java.lang.NoSuchMethodError: 'java.util.SortedMap com.palantir.atlasdb.transaction.api.Transaction.getRows(com.palantir.atlasdb.keyvalue.api.TableReference, java.lang.Iterable, com.palantir.atlasdb.keyvalue.api.ColumnSelection)'
	at com.palantir.atlas.sequential.schema.generated.OffsetsTable.getRows(OffsetsTable.java:613)
	at com.palantir.atlas.sequential.store.OffsetsStore.getOffsets(OffsetsStore.java:79)
	at com.palantir.atlas.sequential.store.OffsetsStore.getLatestOffsets(OffsetsStore.java:40)
	at com.palantir.atlas.sequential.OrderedTableReader.getLatestOffsets(OrderedTableReader.java:95)
	at com.palantir.atlas.sequential.ObjectMappingReader.getLatestOffsets(ObjectMappingReader.java:46)
	at com.palantir.atlas.sequential.api.SequentialTableReader.getLatestOffset(SequentialTableReader.java:81)
	at com.palantir.atlas.eventdriven.AtlasEventReader.lambda$getLatestOffset$1(AtlasEventReader.java:34)
```

**Pull request merged**